daemon/graphdriver/overlay/overlay.go

```diff
@@ -99,6 +99,17 @@ func Init(home string, options []string) (graphdriver.Driver, error) {
 		return nil, graphdriver.ErrNotSupported
 	}
 
+	// check if they are running over btrfs
+	var buf syscall.Statfs_t
+	if err := syscall.Statfs(path.Dir(home), &buf); err != nil {
+		return nil, err
+	}
+
+	if graphdriver.FsMagic(buf.Type) == graphdriver.FsMagicBtrfs {
```

Comment:
This is little confusing :) Message is about using over `ext4`, but you checking for `btrfs`. Maybe I am using `reiser`.
Should we deny to run overlay over any fs which is not `ext4`?